src/core/qgsogcutils.cpp

```diff
@@ -2368,6 +2372,13 @@ QDomElement QgsOgcUtilsExprToFilter::expressionFunctionToOgcFilter( const QgsExp
       const QDomNode geomNode = mDoc.importNode( geomDoc.documentElement(), true );
       otherGeomElem = geomNode.toElement();
     }
+    else if ( otherNode->hasCachedStaticValue() && otherNode->cachedStaticValue().canConvert<QgsGeometry>() )
+    {
+      QgsGeometry geom = otherNode->cachedStaticValue().value<QgsGeometry>();
+      otherGeomElem = QgsOgcUtils::geometryToGML( geom, mDoc, mGMLVersion, mSrsName, mInvertAxisOrientation,
```

Comment:
otherGeomElem should probably be tested to check it is not the empty QDomElement() returned when the geometry cannot be converted to GML